mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -82,15 +82,20 @@ where
         // TODO: optimize for the common case of a single chunk.
         let mut buffer = BytesMut::default();
         while let Some(chunk) = result.next().await {
-            let (offset, body) = chunk?;
-            if offset != buffer.len() as u64 {
-                return Err(DataCacheError::InvalidBlockOffset);
+            match chunk {
+                Ok((offset, body)) => {
+                    if offset != buffer.len() as u64 {
+                        return Err(DataCacheError::InvalidBlockOffset);
+                    }
+                    buffer.extend_from_slice(&body);
+
+                    // Ensure the flow-control window is large enough.
+                    // TODO: review if/when we add a header to the block.
```

Comment:
Outdated TODO?

Reply:
yeah, since we don't store header in the object. removed